src/box/errcode.h

```diff
@@ -38,14 +38,60 @@
 extern "C" {
 #endif
 
+/** Error code field type. */
+enum errcode_field_type {
+	ERRCODE_FIELD_TYPE_CHAR,
+	ERRCODE_FIELD_TYPE_INT,
+	ERRCODE_FIELD_TYPE_UINT,
+	ERRCODE_FIELD_TYPE_LLONG,
+	ERRCODE_FIELD_TYPE_ULLONG,
```

Comment:
Let's also add `LONG` and `ULONG`.

Reply:
Done.

<details>
<summary>diff</summary>

```diff
diff --git a/src/box/errcode.c b/src/box/errcode.c
index dd9f75c3c..ed8e05594 100644
--- a/src/box/errcode.c
+++ b/src/box/errcode.c
@@ -73,6 +73,8 @@
 #define CHAR ERRCODE_FIELD_TYPE_CHAR
 #define INT ERRCODE_FIELD_TYPE_INT
 #define UINT ERRCODE_FIELD_TYPE_UINT
+#define LONG ERRCODE_FIELD_TYPE_LONG
+#define ULONG ERRCODE_FIELD_TYPE_ULONG
 #define LLONG ERRCODE_FIELD_TYPE_LLONG
 #define UULONG ERRCODE_FIELD_TYPE_ULLONG
 #define STRING ERRCODE_FIELD_TYPE_STRING
diff --git a/src/box/errcode.h b/src/box/errcode.h
index c87b89578..1995874be 100644
--- a/src/box/errcode.h
+++ b/src/box/errcode.h
@@ -43,6 +43,8 @@ enum errcode_field_type {
 	ERRCODE_FIELD_TYPE_CHAR,
 	ERRCODE_FIELD_TYPE_INT,
 	ERRCODE_FIELD_TYPE_UINT,
+	ERRCODE_FIELD_TYPE_LONG,
+	ERRCODE_FIELD_TYPE_ULONG,
 	ERRCODE_FIELD_TYPE_LLONG,
 	ERRCODE_FIELD_TYPE_ULLONG,
 	ERRCODE_FIELD_TYPE_STRING,
@@ -90,12 +92,14 @@ struct errcode_record {
 	_(ER_TEST_TYPE_CHAR, 10001,		"Test error", "field", CHAR) \
 	_(ER_TEST_TYPE_INT, 10002,		"Test error", "field", INT) \
 	_(ER_TEST_TYPE_UINT, 10003,		"Test error", "field", UINT) \
-	_(ER_TEST_TYPE_LLONG, 10004,		"Test error", "field", LLONG) \
-	_(ER_TEST_TYPE_ULLONG, 10005,		"Test error", "field", UULONG) \
-	_(ER_TEST_TYPE_STRING, 10006,		"Test error", "field", STRING) \
-	_(ER_TEST_TYPE_MSGPACK, 10007,		"Test error", "field", MSGPACK) \
-	_(ER_TEST_TYPE_TUPLE, 10008,		"Test error", "field", TUPLE) \
-	_(ER_TEST_5_ARGS, 10009,		"Test error", "f1", INT, "f2", INT, "f3", INT, "f4", INT, "f5", INT) \
+	_(ER_TEST_TYPE_LONG, 10004,		"Test error", "field", LONG) \
+	_(ER_TEST_TYPE_ULONG, 10005,		"Test error", "field", ULONG) \
+	_(ER_TEST_TYPE_LLONG, 10006,		"Test error", "field", LLONG) \
+	_(ER_TEST_TYPE_ULLONG, 10007,		"Test error", "field", UULONG) \
+	_(ER_TEST_TYPE_STRING, 10008,		"Test error", "field", STRING) \
+	_(ER_TEST_TYPE_MSGPACK, 10009,		"Test error", "field", MSGPACK) \
+	_(ER_TEST_TYPE_TUPLE, 10010,		"Test error", "field", TUPLE) \
+	_(ER_TEST_5_ARGS, 10011,		"Test error", "f1", INT, "f2", INT, "f3", INT, "f4", INT, "f5", INT) \
 
 #else
 #define TEST_ERROR_CODES(_)
diff --git a/src/box/error.cc b/src/box/error.cc
index c47b84c1a..091315225 100644
--- a/src/box/error.cc
+++ b/src/box/error.cc
@@ -187,6 +187,14 @@ client_error_vformat(struct error *e, va_list ap)
 			error_set_uint(e, r->errfields[i].name,
 				       va_arg(ap, unsigned));
 			break;
+		case ERRCODE_FIELD_TYPE_LONG:
+			error_set_int(e, r->errfields[i].name,
+				      va_arg(ap, long));
+			break;
+		case ERRCODE_FIELD_TYPE_ULONG:
+			error_set_uint(e, r->errfields[i].name,
+				       va_arg(ap, unsigned long));
+			break;
 		case ERRCODE_FIELD_TYPE_LLONG:
 			error_set_int(e, r->errfields[i].name,
 				      va_arg(ap, long long));
diff --git a/test/unit/error.c b/test/unit/error.c
index 407b7e0a1..e01558e3f 100644
--- a/test/unit/error.c
+++ b/test/unit/error.c
@@ -637,7 +637,7 @@ static void
 test_client_error_creation(void)
 {
 	header();
-	plan(21);
+	plan(26);
 
 	/* Test CHAR argument type */
 	const char *s;
@@ -668,6 +668,25 @@ test_client_error_creation(void)
 	e = diag_last_error(diag_get());
 	ok(error_get_uint(e, "field", &u) && u == UINT_MAX);
 
+	/* Test LONG argument type */
+	diag_set(ClientError, ER_TEST_TYPE_LONG, 1);
+	e = diag_last_error(diag_get());
+	ok(error_get_int(e, "field", &i) && i == 1);
+	diag_set(ClientError, ER_TEST_TYPE_LONG, LONG_MAX);
+	e = diag_last_error(diag_get());
+	ok(error_get_int(e, "field", &i) && i == LONG_MAX);
+	diag_set(ClientError, ER_TEST_TYPE_LONG, LONG_MIN);
+	e = diag_last_error(diag_get());
+	ok(error_get_int(e, "field", &i) && i == LONG_MIN);
+
+	/* Test ULONG argument type */
+	diag_set(ClientError, ER_TEST_TYPE_ULONG, 1);
+	e = diag_last_error(diag_get());
+	ok(error_get_uint(e, "field", &u) && u == 1);
+	diag_set(ClientError, ER_TEST_TYPE_ULONG, ULONG_MAX);
+	e = diag_last_error(diag_get());
+	ok(error_get_uint(e, "field", &u) && u == ULONG_MAX);
+
 	/* Test LLONG argument type */
 	diag_set(ClientError, ER_TEST_TYPE_LLONG, 1);
 	e = diag_last_error(diag_get());

```

</details>